tex/alg-NT/classgrp.tex

```diff
@@ -779,6 +836,290 @@ \section{Computation of class numbers}
+Or, in short,
+\[
+	(\text{some matrix $\in \GL_n(\ZZ)$ with determinant $d$}) \cdot
+	\begin{pmatrix} x_1 \\ \vdots \\ x_n \end{pmatrix}
+	= (\text{some vector $\in \ZZ^{\oplus n}$}).
+\]
+\begin{ques}
+	Finish the proof that $d \cdot x \in A$.
+	(Cramer's rule. Or take the adjoint.)
+\end{ques}
+
+Finally, we have that $\OO_K$ is squeezed between two free $\ZZ$-modules of rank $n$, so it is
+also free of rank $n$.\footnote{We did something similar in \Cref{remark:prime_ideals_are_maximal}.}
+\begin{ques}
+	Finish this. (\Cref{thm:fund_fg_abelian_group} can be used here.)
+\end{ques}
+
+\subsection{Second proof}
+% chapter I, section 2, proposition 7 in Lang
+
+This time around, instead of dividing by $d$, we instead take the \emph{dual lattice}.
+
+What is the dual lattice, and why would we think about using it?
+One way to motivate this proof is to look at the inverse of an ideal:
+if $(1) \mid \ka$ (i.e. $\ka$ is an integral ideal), then $\ka\inv \mid (1)$.
+
+Here, $\ka\inv \defeq \{ x \in K \mid x a \in \OO_K \text{ for all }a \in \ka\}$.
+
+We can think of doing something similar, considering
+\[
+	\{ x \in K \mid x a \in \OO_K \text{ for all }a \in \langle \alpha_1, \dots, \alpha_n\rangle \}.
+\]
+This set is actually a lattice of rank $n$, but this won't work to prove the argument!
+We're trying to prove $\OO_K$ is ``discrete'' in the first place, if $\OO_K = K$ then the set above
+would equal $K$ as well.
+
+Instead, we must rely on what we already know --- the discreteness of $\ZZ$.
+Define
+\[
+	S \defeq \{ x \in K \mid \TrK(x a) \in \ZZ \text{ for all }a \in
+	\langle \alpha_1, \dots, \alpha_n\rangle \}.
+\]
+This set is a bit larger than the previous set.
+\begin{ques}
+	Verify that this set is a superset of the previous one. Then conclude that $\OO_K \subseteq S$.
+\end{ques}
+\begin{exercise}
+	Consider $\langle 1, 2i \rangle \subseteq \ZZ[i]$. What would the set $S$ be? (Recall that the
+	trace in $\CC$ is just twice the real part.)
+\end{exercise}
+
+$S$ is still a lattice of rank $n$ --- and this time, we can actually prove it! Since we already
+know $\ZZ$ is discrete.
+
+Now, this is a purely algebraic problem,
+we will only need to use knowledge of vector space here.
+Each element $x \in K$ can be written as a vector
+\[ F(x) \defeq \begin{pmatrix} x_1 \\ \vdots \\ x_n \end{pmatrix} \in \QQ^{\oplus n} \]
+if we use the basis $\{ \alpha_1, \dots, \alpha_n \}$.
+
+\begin{exercise}
+	Show that $(x, y) \mapsto \TrK(x \cdot y)$ can then be written as an \emph{invertible} matrix in
+	$\GL_n(\QQ)$ ---
+	specifically, there exists $M \in \GL_n(\QQ)$ such that $\TrK(x \cdot y) = F(x)^\top M F(y)$,
+	where $F(x)^\top$ is the transpose of $F(x)$.
+\end{exercise}
+
+Which makes $(x, y) \mapsto \TrK(x \cdot y)$ \emph{almost} an inner product
+(see \Cref{ch:inner_product_spaces}),
+except that it is not positive definite (for example, in $\CC$, we have $\Tr((1+i)^2) = 0$).
+But having the matrix invertible suffices to do the following:
```

Comment:
This "feels" a bit silly that we're proving it the "wrong" way (the book so throughoutly uses the basis-less ("Math 55a") approach; in fact before this pull request there's exactly one more position in the book that transposes a matrix instead of taking the dual of a linear transformation), but for some reason the normal proof (the bilinear form (x, y) ↦ Tr(xy) is nondegenerate so we're done) doesn't really sink in for me, and Napkin didn't define what is a nondegenerate bilinear form anyway.